cmd/juju/application/updateseries.go

```diff
@@ -0,0 +1,187 @@
+// by the update-series command.
+type charmAPI interface {
+	CharmInfo(charmURL string) (*charms.CharmInfo, error)
+}
+
+// updateSeriesCommand is responsible for updating the series of an application or machine.
+type updateSeriesCommand struct {
+	modelcmd.ModelCommandBase
+
+	apiRoot            api.Connection
+	charmClient        charmAPI
+	updateSeriesClient updateSeriesAPI
+
+	applicationName string
+	force           bool
+	machineNumber   string
+	series          string
+}
+
+var updateSeriesDoc = `
+When no flags are set, an application or machines series value with be updated
+within juju.
+
+The update is disallowed unless the --force flag is used if the requested
+series is not explicitly supported by the application's charm and all
+subordinates, as well as any other charms which may be deployed to the same
+machine.
+
+In the case of updating a machine's series, the --force option should be used
+with caution since using a charm on a machine running an unsupported series may
+cause unexpected behavior.  Alternately, if the requested series supported in
+later revisions of the charm, upgrade-charm can run beforehand.
+
+Examples:
+	juju update-series <application> <series>
+	juju update-series <application> <series> --force
+	juju update-series <machine> <series>
+	juju update-series <machine> <series> --force
+
+See also:
+    machines
+    status
+    upgrade-charm
+`
+
+func (c *updateSeriesCommand) Info() *cmd.Info {
+	return &cmd.Info{
+		Name:    "update-series",
+		Args:    "[<application>|<machine>] <series>",
+		Purpose: "Update an application or machine's series.",
+		Doc:     updateSeriesDoc,
+	}
+}
+
+func (c *updateSeriesCommand) SetFlags(f *gnuflag.FlagSet) {
+	c.ModelCommandBase.SetFlags(f)
+	f.BoolVar(&c.force, "force", false, "Update even if the series is not supported by the charm and/or related subordinate charms.")
+}
+
+// Init implements cmd.Command.
+func (c *updateSeriesCommand) Init(args []string) error {
+	switch len(args) {
+	case 2:
+		if names.IsValidApplication(args[0]) {
+			c.applicationName = args[0]
+		} else if names.IsValidMachine(args[0]) {
+			c.machineNumber = args[0]
+		} else {
+			return errors.Errorf("invalid application or machine name %q", args[0])
+		}
+		c.series = args[1]
+	case 1:
+		if names.IsValidMachine(args[0]) {
+			return errors.Errorf("no series specified")
+		}
+		return errors.Errorf("no application name or no series specified")
+	case 0:
+		return errors.Errorf("no arguments specified")
+	default:
+		return cmd.CheckEmpty(args[2:])
+	}
+	return nil
+}
+
+// Run implements cmd.Run.
+func (c *updateSeriesCommand) Run(ctx *cmd.Context) error {
+
+	if c.updateSeriesClient == nil && c.charmClient == nil {
+		apiRoot, err := c.NewAPIRoot()
+		if err != nil {
+			return errors.Trace(err)
+		}
+		c.apiRoot = apiRoot
+		c.updateSeriesClient = application.NewClient(c.apiRoot)
+		c.charmClient = charms.NewClient(c.apiRoot)
+	}
+
+	if c.apiRoot.BestFacadeVersion("Application") < 5 {
```

Comment:
This was copied from elsewhere I realise, but it's better to use the BestAPIVersion() method on c.updateSeriesClient
You don't need to retype the facade name as a string and it's easier to mock for testing.